pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -100,6 +101,42 @@ type InfoItem struct {
 
 	// Value is human readable content.
 	Value string `json:"value,omitempty"`
+
+	// Value is derived from another source.
+	ValueFrom *InfoItemSource `json:"valueFrom,omitempty"`
+}
+
+type InfoItemSource struct {
+  SecretKeyRef *ConfigMapKeySelector `json:"secretRef,omitempty"`
```

Comment:
could we not just reuse `core.SecretKeySelector`?

Reply:
Thanks for the pointer. Yes, we can re-use it. Updated!